### DISCUSSION TITLE
Is this package compatible with CI/CD?

### DISCUSSION THREAD

**Discussion by TheDevick** [paratestphp/paratest#817] [Q&A]
It's may a dumb question, but can we use it on Github Workflows ( With no big issues )?

If yes, it's there any example of a `.yml` file for CI on Github Actions?

**Comment by mfn** [comment]
It works like anything else you write a github action workflow: make sure your PHP version is right, composer dependencies are installed and call `vendor/bin/paratest`, assuming you configured it correctly with it's default options or pass additional args.

I usually wrap them in a composer `scripts` entry and call this in CI/CD, so I'm sure when I call it locally, it's the same args.

**Comment by santigarcor** [comment]
@mfn I tried running it on GitHub Actions but it takes the same amount of time than the normal phpunit run. Did you do anything special?

**Reply by antonvolokha** [reply]
I encountered the same issue and found that the free/default GitHub Actions runner has only 2 cores. Paratest creates one process per core, so using this library on GitHub Actions doesn't make sense if you're expecting performance improvements. To see a noticeable performance boost, you would need to purchase a larger runner or set up your own custom runner.

**Reply by Slamdunk** [reply]
The difference in running ParaTest vs PHPUnit in the default, free Github runner is indeed small compared to the gain you get on you local 12 or 16 cores CPU.
Still I suggest to keep using ParaTest in CI too for two reasons:

1. The boost is small, but present nonetheless. Here's the difference for our main project:
    ```diff
    -Time: 03:32.804
    +Time: 03:05.230
    ```
2. Local and CI build shall be as similar as possible
